BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/rootAnalysis.py

```diff
@@ -1,297 +1,238 @@
+    #interp_cross = np.interp(E_EV,cross_section_E,cross_sections) #cm^2
+    
+    IceVolume = 2.68592e7 #km^3
+    #rho_factor=917.0/1000.00 #g/cm^3
+    #m_n = 1.67e-24 #g
+
+    #L_int = m_n/(rho_factor*interp_cross)/100 #m
+    #interaction_lengths = L_int
+
+    effective_V = []
+
+    effective_V_p = []
+    effective_V_m = []
+
+    #all_weighted = []
+    #all_passed = []
+    #all_events = []
+    #all_weights_squared = []
+
+    #print('E_EV: ', E_EV)
+    #print('PassingWeights: ', PassingWeights)
+    #print('PassingEvents: ', PassingEvents)
+    #print('TotalEvents: ', TotalEvents)
+
+    for en in E_EV:
+
+        total_weighted = np.sum(PassingWeights[en])
+        #total_passed = np.sum(PassingEvents[en])
+        total_events = np.sum(TotalEvents[en])
+
+        #all_weighted.append(total_weighted)
+        #all_passed.append(total_passed)
+        #all_events.append(total_events)
+        #all_weights_squared.append(np.sum(np.asarray(PassingWeights[en])**2))
+
+
+        #print('at energy ', en, ', ', total_weighted,total_passed,'events passed out of ',total_events)
+        #print('average rawWeights is : ',np.mean(RawWeights[en]))
+        error_p, error_m = AddErrors(RawWeights[en])
+        #print(total_weighted,events_error,total_weighted*frac_error_p,total_weighted*frac_error_m)
+
+        effective_V.append(IceVolume*total_weighted/total_events*4*np.pi)
+        #print("effective V is ", IceVolume*total_weighted/total_events*4*np.pi)
+        #print("components were", IceVolume, total_weighted, total_events)
+        effective_V_p.append(IceVolume*(error_p)/total_events*4*np.pi)
+        effective_V_m.append(IceVolume*(error_m)/total_events*4*np.pi)
+        #effective_A.append(effective_V[-1]/(L_int/100000))#factor of 100000 to convert from cm to km 
+
+    #all_weighted = np.asarray(all_weighted)
+    #all_passed = np.asarray(all_passed)
+    #all_events = np.asarray(all_events)
+
+    effective_V = np.asarray(effective_V)
+    effective_V_p = np.asarray(effective_V_p)
+    effective_V_m = np.asarray(effective_V_m)
+
+
+    with open(g.opath+"/"+str(g.gen)+"_pueoOut.csv",'a') as f:
+        f.write(str(g.indiv)+","+str(effective_V[0])+","+str(effective_V_p[0])+","+str(effective_V_m[0])+"\n")
+        print("Writing to:", g.opath+"/"+str(g.gen)+"_pueoOut.csv" )
+
+    with open(g.opath+"/"+str(g.gen)+"_fitnessScores.csv",'a') as f:
+        f.write(str(effective_V[0])+"\n")
+        print("Writing to:", g.opath+"/"+str(g.gen)+"_fitnessScores.csv")
+
+    with open(g.opath+"/"+str(g.gen)+"_vEffectives.csv",'a') as f:
+        f.write(str(effective_V[0])+"\n")
+        print("Writing to:", g.opath+"/"+str(g.gen)+"_fitnessScores.csv")
+
+    with open(g.opath+"/"+str(g.gen)+"_errorBars.csv",'a') as f:
+        f.write(str(effective_V_p[0])+","+str(effective_V_m[0])+"\n")
+        print("Writing to:", g.opath+"/"+str(g.gen)+"_errorBars.csv")
```

Comment:
This can be simplified using `pathlib` and/or f-strings.

Reply:
converted to .format()